### ISSUE TITLE
Occasional Crash over “Required status check”

### ISSUE THREAD

**Issue by tiagolobocastro** [bors-ng/bors-ng#1150]
Hi,

We use bors.tech with our github and occasionally we see a crash:
```
{{:badmatch,
  {:error, :push, 422,
   "{\"message\":\"Required status check \\\"commitlint\\\" is expected.\",\"documentation_url\":\"https://docs.github.com/articles/about-protected-branches\"}"}},
 [
   {BorsNG.Worker.Batcher, :complete_batch, 3,
    [file: 'lib/worker/batcher.ex', line: 712]},
   {BorsNG.Worker.Batcher, :maybe_complete_batch, 1,
    [file: 'lib/worker/batcher.ex', line: 658]},
   {BorsNG.Worker.Batcher, :handle_cast, 2,
    [file: 'lib/worker/batcher.ex', line: 94]},
   {:gen_server, :try_dispatch, 4,
    [file: 'gen_server.erl', line: 637]},
   {:gen_server, :handle_msg, 6,
    [file: 'gen_server.erl', line: 711]},
   {:proc_lib, :init_p_do_apply, 3,
    [file: 'proc_lib.erl', line: 249]}
 ]}
```

That `commmitlint` is indeed a required status, though not sure why it's causing the crash.
Not sure if related to this or not but we've also observed that sometimes the PR's get stuck and we have to rebase, force push and retry, though since the crash does not mention a specific build/pr not sure how to correlate those things.

Any clues what could be the issue here?
Thank you!

**Comment by jkryl** [comment]
I have the same problem (on the same project). It happened in the second run of tests when bors was trying to bisect commits. The second run on subset of PRs passed and therefore the PR should have been merged but bors crashes instead. Of course the commitlint check passed so even the message about required status check is wrong.

**Comment by tommilligan** [comment]
Hey @tiagolobocastro , I believe this can be now closed as fixed by #1273. Please take a look and verify if it covers your use case, or provide more details if not.

The crash is caused when bors tries to merge a commit into your master branch, on which there is branch protection setup (and not fulfilled). My guess is that the `commitlint` check is still running on the new commit that bors created out of a subset of commits, at the time bors tries to merge the commit.

#1273 probably won't get the commit merging in in this case, but at least will provide a clear error message and not crash.

**Comment by tiagolobocastro** [comment]
Brilliant thanks @tommilligan :+1: I'll close this one then. (can always reopen if it happens again)
